src/material-experimental/mdc-checkbox/harness/checkbox-harness.ts

```diff
@@ -0,0 +1,128 @@
+    return new HarnessPredicate(MatCheckboxHarness)
+        .addOption('label', options.label,
+            (harness, label) => HarnessPredicate.stringMatches(harness.getLabelText(), label));
+  }
+
+  private _label = this.locatorFor('.mat-checkbox-label');
+  private _input = this.locatorFor('input');
+
+  /** Gets a boolean promise indicating if the checkbox is checked. */
+  async isChecked(): Promise<boolean> {
+    const checked = (await this._input()).getAttribute('checked');
+    return coerceBooleanProperty(await checked);
+  }
+
+  /** Gets a boolean promise indicating if the checkbox is in an indeterminate state. */
+  async isIndeterminate(): Promise<boolean> {
+    const indeterminate = (await this._input()).getAttribute('indeterminate');
+    return coerceBooleanProperty(await indeterminate);
+  }
+
+  /** Gets a boolean promise indicating if the checkbox is disabled. */
+  async isDisabled(): Promise<boolean> {
+    const disabled = (await this._input()).getAttribute('disabled');
+    return coerceBooleanProperty(await disabled);
+  }
+
+  /** Gets a boolean promise indicating if the checkbox is required. */
+  async isRequired(): Promise<boolean> {
+    const required = (await this._input()).getAttribute('required');
+    return coerceBooleanProperty(await required);
+  }
+
+  /** Gets a boolean promise indicating if the checkbox is valid. */
+  async isValid(): Promise<boolean> {
+    const invalid = (await this.host()).hasClass('ng-invalid');
+    return !(await invalid);
+  }
+
+  /** Gets a promise for the checkbox's name. */
+  async getName(): Promise<string | null> {
+    return (await this._input()).getAttribute('name');
+  }
+
+  /** Gets a promise for the checkbox's value. */
+  async getValue(): Promise<string | null> {
+    return (await this._input()).getAttribute('value');
+  }
+
+  /** Gets a promise for the checkbox's aria-label. */
+  async getAriaLabel(): Promise<string | null> {
+    return (await this._input()).getAttribute('aria-label');
+  }
+
+  /** Gets a promise for the checkbox's aria-labelledby. */
+  async getAriaLabelledby(): Promise<string | null> {
+    return (await this._input()).getAttribute('aria-labelledby');
+  }
+
+  /** Gets a promise for the checkbox's label text. */
+  async getLabelText(): Promise<string> {
+    return (await this._label()).text();
+  }
+
+  /** Focuses the checkbox and returns a void promise that indicates when the action is complete. */
+  async foucs(): Promise<void> {
+    return (await this._input()).focus();
+  }
+
+  /** Blurs the checkbox and returns a void promise that indicates when the action is complete. */
+  async blur(): Promise<void> {
+    return (await this._input()).blur();
+  }
+
+  /**
+   * Toggle the checked state of the checkbox and returns a void promise that indicates when the
+   * action is complete.
+   */
+  async toggle(): Promise<void> {
+    return (await this._input()).click();
```

Comment:
There might be some unexpected behavior here with `MAT_CHECKBOX_CLICK_ACTION`, where clicking doesn't _necessarily_ toggle the checked state

(same applies for the mdc one)

Reply:
Added docs explaining this